lib/ops/opsservice/site.go

```diff
@@ -604,6 +605,49 @@ func (s *site) getClusterEnvironmentVariables() (env storage.EnvironmentVariable
 	return env, nil
 }
 
+// updateServiceConfiguration updates the gravity-site service configuration if
+// the gravityControllerService configuration has been modified.
+func (s *site) updateServiceConfiguration() error {
```

Comment:
Seems like this function will always try to update the service as long as gravityController is present in the resource (even if it hasn't changed)? I think it should only be called if there really was a change in the gravity-site settings when a user updates the clusterconfig resource.